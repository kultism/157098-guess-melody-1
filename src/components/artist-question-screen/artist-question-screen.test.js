import React from 'react';
import ArtistQuestionScreen from './artist-question-screen';
import renderer from 'react-test-renderer';

const question = {
  type: `artist`,
  song: {
    artist: `One`,
    src: `path`,
  },
  answers: [
    {
      picture: `path`,
      artist: `One`,
    },
    {
      picture: `path`,
      artist: `Two`,
    },
    {
      picture: `path`,
      artist: `Three`,
    }
  ],
};

describe(`<ArtistQuestionScreen/>`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(<ArtistQuestionScreen onAnswer={jest.fn()} question={question}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
