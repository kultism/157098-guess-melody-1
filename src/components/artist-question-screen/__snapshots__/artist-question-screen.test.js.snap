// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistQuestionScreen/> renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio
        src="path"
      />
    </div>
  </div>
  <form
    className="game__artist"
    onChange={[MockFunction]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="One"
          className="artist__picture"
        />
        One
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Two"
          className="artist__picture"
        />
        Two
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Three"
          className="artist__picture"
        />
        Three
      </label>
    </div>
  </form>
</section>
`;
